### PR TITLE
Support for conda packaging

### PR DESCRIPTION
**Pull request by bjlittle**
@pletzer This pull-request adds some necessary changes to support the task of packaging `mint` for `conda-forge`.

Primarily:

- the `pymint` directory is now moved to `mint`, which is the name of the importable Python package.
- the `setup.py` uses `Cython` to build the `mint` package, which now allows you to now do all of the following:
  ```python
  python setup.py build
  python setup.py install
  python setup.py develop
  python -m pip install .
  python -m pip install --editable .
  ```

In addition, the `requirements/mint.yml` file now contains all the developer package dependencies for `mint` (and the `README.md` has also been updated to reflect this). Essentially, creating a `mint` developer environment simply involves:
```shell
> conda env create --file requirements/mint.yml
> . activate mint-dev
```

### CODE REVIEW

**Comment by bjlittle** [comment]
@pletzer Once you've reviewed this and are happy to merge to `master`.

Then could you please create a `v1.8.0` release, akin to https://github.com/bjlittle/mint/releases/tag/v1.8.0

Would it be okay to follow [PEP-0440](https://www.python.org/dev/peps/pep-0440/) versioning? Otherwise, this is most likely to cause issues with infra-structure tooling.

**Comment by bjlittle** [comment]
Once this PR is merged and you've cut a `v1.8.0` release, I can then reference the GitHub release tag in the conda recipe.

This will involve creating a `conda-forge` feedstock for `python-mint`, which is the package name.

To install the package you would do `conda install -c conda-forge python-mint`.

However to use the installed `python-mint` package you would simply do `python -c "import mint"`

:smile:

File: README.md [hunk 1]
```diff
@@ -11,48 +11,49 @@ loop integrals of an interpolated vector field deriving from a gradient is zero.
...
-## How to build the python python-mint module
-
-The MINT python interface (python-mint) requires VTK, netCDF and the tbb libraries to 
+The MINT Python interface requires VTK, netCDF and the tbb libraries to 
 be installed. This is most easily done in a conda environment:
 ```
-conda create -n mintenv python=3.9 vtk=9.0.1 pytest
-conda activate mintenv
+conda env create --file requirements/mint.yml
+conda activate mint-dev
```
Comment (bjlittle) [inline README.md:38]: This is pretty neat, huh :wink:

File: requirements/mint.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
+name: mint-dev
```
Comment (bjlittle) [inline requirements/mint.yml:1]: This is the name of the environment that `conda env` will create.

File: setup.py [hunk 3]
```diff
@@ -93,16 +130,10 @@ def getCondaNetcdf():
...
-    ext_modules = [setuptools.Extension('libmint', # name of the shared library
-                   sources=glob.glob('src/*.cpp'),
-                   define_macros=[],
-                   include_dirs=['src/', vtklib["VTK_INCLUDE_DIR"], nclib["NETCDF_INCLUDE_DIR"]],
-                   libraries=vtklib["VTK_LIBRARIES"] + nclib["NETCDF_LIBRARIES"],
-                   library_dirs=[ vtklib["VTK_INCLUDE_DIR"], nclib["NETCDF_LIBRARIES_DIR"] ],
-                   extra_compile_args=[cpp11_flag,],
-                   ),],
+    packages=[PACKAGE],
+    ext_modules=cythonize(extensions),
```
Comment (bjlittle) [inline setup.py:134]: I hope using `cython` isn't a deal breaker?

**Comment by bjlittle** [comment]
@pletzer Would it be possible for you to checkout my pull-request branch and build `mint` in a conda environment on your OSX? Just to confirm whether you can recreate the OSX build error that we're seeing on `conda-forge`, see https://github.com/conda-forge/staged-recipes/pull/14354

**Comment by pletzer** [comment]
Thanks a lot @bjlittle. Nice to see a successful build on Linux and Windows!
* Not sure what the build problem on Mac OS X, it seems to be related to division by zero in preprocessor expression. Maybe related to _FORTIFY_SOURCE? I don't get that error on my Mac. 
* pytest is no longer working for. I get 
* (mint-dev) NIWA-1019500~/bjlittle-mint/mint/tests$ pytest 
=============================================== test session starts ================================================
platform darwin -- Python 3.9.2, pytest-6.2.2, py-1.10.0, pluggy-0.13.1
rootdir: /Users/pletzera/bjlittle-mint
collected 6 items                                                                                                  

test_grid.py FFException ignored in: <function Grid.__del__ at 0x109ab2af0>
Traceback (most recent call last):
  File "/Users/pletzera/bjlittle-mint/mint/grid.py", line 31, in __del__
    LIB.mnt_grid_del.argtypes = [POINTER(c_void_p)]
AttributeError: 'NoneType' object has no attribute 'mnt_grid_del'
Exception ignored in: <function Grid.__del__ at 0x109ab2af0>
Traceback (most recent call last):
  File "/Users/pletzera/bjlittle-mint/mint/grid.py", line 31, in __del__
    LIB.mnt_grid_del.argtypes = [POINTER(c_void_p)]
AttributeError: 'NoneType' object has no attribute 'mnt_grid_del'

It looks like the code cannot find the shared library.... To investigate.

**Comment by pletzer** [comment]
Strangely enough the tests run manually, just not through pytest...

**Comment by bjlittle** [comment]
@pletzer I'm skipping the build for Windows

I think the issue is that the `setup.py` should be using `CXXFLAGS` and not `CPPFLAGS`

**Comment by bjlittle** [comment]
I'm going to experiment with this just now... and let you know.

**Comment by bjlittle** [comment]
Do you think that you could enable `GitHub Discussions` on your repo? Then we could chat easier perhaps.

For your repo, simply go to `Settings`, then scroll down to the `Features` section, and it's the last option to checkbox.

I've enabled it for `iris`, see https://github.com/SciTools/iris/discussions

**Comment by bjlittle** [comment]
Also, I think there is a missing pytest configuration... I'll also look into that too

**Comment by pletzer** [comment]
Thanks @bjlittle !

I accepted you PR but needed to call the directory mint/ something else, python_mint for the time being, in order for pytest to run (?).

**Comment by bjlittle** [comment]
@pletzer Awesome thanks! :beers: 

As it happens there is no need to do that. It's simply that `pytest` requires to be configured in a different way since I removed the `package_dir` directive from the `setup.py`.

I'll push another pull-request, and you'll see what I mean :smile:

**Comment by bjlittle** [comment]
@pletzer Also I think I've just cracked my `osx` building issue :+1: 

I'm testing on `conda-forge`... but it's looking really good!

**Comment by pletzer** [comment]
Hi @bjlittle 
I don't see how I can make you administrator, see https://docs.github.com/en/github/setting-up-and-managing-your-github-user-account/permission-levels-for-a-user-account-repository

**Comment by pletzer** [comment]
@bjlittle looks like we're complying with standard versioning, let me know otherwise.

**Comment by pletzer** [comment]
@bjlittle How's https://github.com/pletzer/mint/releases/tag/v1.8.0 ?

**Pull request merged**